02-microarray/ortholog-mapping_microarray_01_ensembl.Rmd

```diff
@@ -0,0 +1,390 @@
+
+
+# Ortholog Mapping with `hcop` - Microarray
+
+## Install libraries
+
+See our Getting Started page with [instructions for package installation](https://alexslemonade.github.io/refinebio-examples/01-getting-started/getting-started.html#what-you-need-to-install) for a list of the other software you will need, as well as more tips and resources.
+
+In this analysis, we will be using `hcop` to map between zebrafish and human identifiers [@Turner2018].
+
+We'll need to use the `devtools` package to install `hcop` from GitHub [@Wickham2020].
+
+```{r}
+if (!("devtools" %in% installed.packages())) {
+  install.packages("devtools")
+}
+```
+
+Now we can install `hcop` from Github.
+You can control what version of the package is installed by using the `ref` argument of `devtools::install_github()`.
+Here, we'll use the most recent commit at the time that we are putting together this example.
+
+```{r}
+devtools::install_github("stephenturner/hcop", ref = "0985fddc91a6ef2308f4800958dfd11c25fe6a98")
+```
+
+Attach the packages we need for this analysis.
+
+```{r}
+# Attach the library
+library(hcop)
+
+# We will need this so we can use the pipe: %>%
+library(magrittr)
+```
+
+## Import and set up data
+
+Data downloaded from refine.bio include a metadata tab separated values (TSV) file and a data TSV file. 
+This chunk of code will read the both TSV files and add them as data frames to your environment.
+
+```{r}
+# Read in metadata TSV file
+metadata <- readr::read_tsv(file.path(
+  data_dir, # Replace with path to your metadata file
+  "metadata_GSE71270.tsv" # Replace with the name of your metadata file
+))
+
+# Read in data TSV file
+df <- readr::read_tsv(file.path(
+  data_dir, # Replace with path to your data file
+  "GSE71270.tsv" # Replace with the name of your data file
+)) %>%
+  # Tuck away the Gene ID column as rownames
+  tibble::column_to_rownames("Gene")
+```
+
+Let's ensure that the metadata and data are in the same sample order. 
+
+```{r}
+# Make the data in the order of the metadata
+df <- df %>%
+  dplyr::select(metadata$geo_accession)
+
+# Check if this is in the same order
+all.equal(colnames(df), metadata$geo_accession)
+
+# Bring back the "Gene" column in preparation for mapping
+df <- df %>%
+  tibble::rownames_to_column("Gene")
+```
+
+## Mapping human gene symbols to zebrafish Ensembl gene IDs
+
+refine.bio data uses Ensembl gene identifiers, which will be in the first column.
+
+```{r}
+# Let's take a look at the first 6 items in the `Gene` column of `df`
+head(df$Gene)
+```
+
+Ensembl gene identifiers have different species-specific prefixes. 
+In zebrafish, Ensembl gene identifiers begin with `ENSDARG` (in human, `ENSG`, etc.).
+
+Now let's do the mapping!
+We'll use the `zebrafish` data from `hcop`; this is what `zebrafish` looks like.
+
+```{r}
+zebrafish
+```
+
+We're interested in the `human_symbol` and `zebrafish_ensembl` columns specifically.
```

Comment:
The power of `hcop` is that it is a compilation of multiple orthology predctions. Because of that, one of the most important parts of the data is the `support` column, so we should make use of that as well. This suggests a way to resolve some of the multimappings: take the one with the most support, or with support from the databases you trust. (Most support is probably the best choice for this notebook.)